File: sol/solution/0001-0100/0008. String to Integer (atoi)/Solution.go
func myAtoi(s string) int {
    i, sign, result := 0, 1, 0
    n := len(s)
    for i < n && s[i] == ' ' { i++ }
    if i < n && (s[i] == '+' || s[i] == '-') {
        if s[i] == '-' { sign = -1 }
        i++
    }
    for i < n && s[i] >= '0' && s[i] <= '9' {
        digit := int(s[i] - '0')
        if result > (1<<31 - 1 - digit)/10 {
            if sign == 1 { return 1<<31 - 1 } else { return -1 << 31 }
        }
        result = result * 10 + digit
        i++
    }
    return result * sign
}

File: sol/solution/0001-0100/0008. String to Integer (atoi)/Solution.php
class Solution {
    /**
     * @param String $s
     * @return Integer
     */
    function myAtoi($s) {
        $s = ltrim($s); 
        $n = strlen($s);
        $i = 0;
        $sign = 1;
        $result = 0;
        if ($i < $n && ($s[$i] === '-' || $s[$i] === '+')) {
            $sign = ($s[$i] === '-') ? -1 : 1;
            $i++;
        }
        while ($i < $n && ctype_digit($s[$i])) {
            $digit = intval($s[$i]);
            if ($result > (2147483647 - $digit) / 10) {
                return $sign === 1 ? 2147483647 : -2147483648;
            }
            $result = $result * 10 + $digit;
            $i++;
        }
        return $sign * $result;
    }
}